privacyidea/lib/security/default.py

```diff
@@ -325,25 +325,27 @@ def password_encrypt(text, password):
         """
         Encrypt the given text with the password.
         A key is derived from the password and used to encrypt the text in
-        AES MODE_CBC. The IV is returned togeather with the cipher text.
-        <IV:Ciphter>
+        AES MODE_CBC. The IV is returned together with the cipher text.
+        <IV:Cipher>
 
         :param text: The text to encrypt
+        :type text: str
```

Comment:
I wasn't really aware of this method :-) But I see it's used to encrypt encryption keys with a password in ``encrypt_enckey`` in ``pi-manage``. I think encryption keys are just bytes. What do you think about making ``password_encrypt`` take bytes and return bytes, and ``password_decrypt`` take bytes and return bytes, in order to make it consistent with the other methods of ``DefaultSecurityModule``?

Also, I think we'll need to change ``encrypt_enckey`` in ``pi-manage``: it reads the enckey file using ``open(encfile, "r")`` which I think will break in Python 3?

Reply:
I stumbled upon this as well, we need to distinguish between functions that expect/return strings (hashing, encrypting/decrypting passwords/pins) and functions which handle arbitrary (binary) data. I am on it.

And there are several `open()` calls which need to be changed to read the data in binary format. I'll fix them along the way.

Reply:
This is all a little tricky: `encrypt_enckey` in `pi-manage` reads in an encKey file (which should be binary) and prints out a string (hexlified iv and hexlified encrypted data).
This string will be read in by `_get_secret()` in the `DefaultSecurityModule` which then calls `password_decrypt` with the given string.
I am not really sure how we should handle things here.
We are going to rewrite these parts anyway when we switch to a different crypto-library (AES, padding, etc...), so maybe we should keep it that way for now and keep this in mind for the next rewrite.

Reply:
I think adding a TODO at those functions should be fine.

Reply:
:+1: